universal-application-tool-0.0.1/test/services/applicant/BlockTest.java

```diff
@@ -0,0 +1,149 @@
+package services.applicant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.OptionalLong;
+import org.junit.Test;
+import services.Path;
+import services.program.BlockDefinition;
+import services.program.ProgramQuestionDefinition;
+import services.question.NameQuestionDefinition;
+import services.question.QuestionDefinition;
+import services.question.TextQuestionDefinition;
+
+public class BlockTest {
+
+  @Test
+  public void create() {
+    BlockDefinition definition =
+        BlockDefinition.builder().setId(123L).setName("name").setDescription("description").build();
+    Block block = Block.create(1L, definition, new ApplicantData());
+    assertThat(block.id()).isEqualTo(1L);
+    assertThat(block.getName()).isEqualTo("name");
+    assertThat(block.getDescription()).isEqualTo("description");
+    assertThat(block.getQuestions()).isEmpty();
+    assertThat(block.hasErrors()).isFalse();
+    assertThat(block.isComplete()).isTrue();
+  }
+
+  @Test
+  public void getQuestions_returnsCorrectApplicantQuestions() {
+    QuestionDefinition name =
+        new NameQuestionDefinition(
+            OptionalLong.of(1L),
+            1L,
+            "name",
+            Path.empty(),
+            "",
+            ImmutableMap.of(),
+            ImmutableMap.of());
+    QuestionDefinition text =
+        new TextQuestionDefinition(
+            OptionalLong.of(2L), 1L, "", Path.empty(), "", ImmutableMap.of(), ImmutableMap.of());
+    BlockDefinition definition =
+        BlockDefinition.builder()
+            .setId(20L)
+            .setName("")
+            .setDescription("")
+            .addQuestion(ProgramQuestionDefinition.create(name))
+            .addQuestion(ProgramQuestionDefinition.create(text))
+            .build();
+    ApplicantData applicantData = new ApplicantData();
+
+    Block block = Block.create(1L, definition, applicantData);
+
+    ImmutableList<ApplicantQuestion> expected =
+        ImmutableList.of(
+            new ApplicantQuestion(name, applicantData), new ApplicantQuestion(text, applicantData));
+    assertThat(block.getQuestions()).containsExactlyElementsOf(expected);
+  }
+
+  @Test
+  public void hasErrors_returnsFalseIfBlockHasNoQuestions() {
+    BlockDefinition definition =
+        BlockDefinition.builder().setId(123L).setName("name").setDescription("description").build();
+    Block block = Block.create(1L, definition, new ApplicantData());
+
+    assertThat(block.hasErrors()).isFalse();
+  }
```

Comment:
Question (if-minor): can we add more tests for hasErrors? It isn't immediately clear what it does though.